### DISCUSSION TITLE
preventing a user to access a route that he/she is not the owner

### DISCUSSION THREAD

**Discussion by miladazhdehnia** [adonisjs/core#2114] [Help] [answered]
hi!
the problem is other users can edit a post that created by other user or even each other properties!
imagine we have a post model and it "blongsTo" a user but other users can edit it with put method, but they must be disabled to access this route.
can fix this with just one middleware?
if not, what is the cleanest way?

**Comment by Jakobud** [comment]
When accessing a route to a specific resource ID you need to run a query to determine if that current user ID has access to it or not. If not, redirect them.

**Comment by iagobruno** [comment, marked as answer]
You can check if authenticated user is authorized to perform an action inside the controller, something like this:

```ts
// app/Controllers/Http/PostsController.ts

export default class PostsController {
  // ...

  public async update ({ auth, params, request, response }) {
    const data = request.all()
    const post = await Post.find(params.id)

    if (post.userId !== auth.user.id) {
      // The logged in user is not the owner
      return response.forbidden()
    }

    post.merge(data)
    await post.save()

    return response.ok()
  }

  // ...
}
```
